chain/client/src/stateless_validation/partial_witness/partial_witness_actor.rs

```diff
@@ -819,11 +819,16 @@ impl PartialWitnessActor {
 }
 
 fn compress_witness(witness: &ChunkStateWitness) -> Result<EncodedChunkStateWitness, Error> {
-    let shard_id_label = witness.chunk_header.shard_id().to_string();
+    let shard_id_label = witness.chunk_header().shard_id().to_string();
     let encode_timer = near_chain::stateless_validation::metrics::CHUNK_STATE_WITNESS_ENCODE_TIME
         .with_label_values(&[shard_id_label.as_str()])
         .start_timer();
-    let (witness_bytes, raw_witness_size) = EncodedChunkStateWitness::encode(witness)?;
+    let (witness_bytes, raw_witness_size) = if let ChunkStateWitness::V1(witness) = witness {
+        // For V1 witness, we need to encode only the inner witness struct for backwards compatibility.
+        EncodedChunkStateWitness::encode(witness)?
+    } else {
+        EncodedChunkStateWitness::encode(witness)?
+    };
```

Comment:
```suggestion
    let (witness_bytes, raw_witness_size) = if let ChunkStateWitness::V1(witness_v1) = witness {
        // For V1 witness, we need to encode only the inner witness struct for backwards compatibility.
        EncodedChunkStateWitness::encode(witness_v1)?
    } else {
        EncodedChunkStateWitness::encode(witness)?
    };
```
otherwise I thought at first that logic is identical :)